src/widgets/chart.rs

```diff
@@ -342,12 +360,25 @@ impl<'a> Chart<'a> {
                 && legend_width < max_legend_width
                 && legend_height < max_legend_height
             {
-                layout.legend_area = Some(Rect::new(
-                    layout.graph_area.right() - legend_width,
-                    layout.graph_area.top(),
-                    legend_width,
-                    legend_height,
-                ));
+                let (x, y) = match self.legend_position {
+                    Corner::TopRight => (
+                        layout.graph_area.right() - legend_width,
+                        layout.graph_area.top(),
+                    ),
+                    Corner::TopLeft => (
+                        layout.graph_area.left(),
+                        layout.graph_area.top() - legend_height
+                            + if self.y_axis.title.is_some() { 1 } else { 0 },
+                    ),
+                    Corner::BottomRight => (
+                        layout.graph_area.right() - legend_width,
+                        layout.graph_area.bottom()
+                            - legend_height
+                            - if self.x_axis.title.is_some() { 1 } else { 0 },
+                    ),
+                    Corner::BottomLeft => (layout.graph_area.left(), layout.graph_area.bottom()),
+                };
+                layout.legend_area = Some(Rect::new(x, y, legend_width, legend_height));
```

Comment:
Refactoring suggestion: pull this chunk out into a method that just calculates the position of the legend (for making the code easier to read.